tooling/miniutil/src/build/terminator.rs

```diff
@@ -13,117 +13,167 @@ impl FunctionBuilder {
+    }
+
     pub fn panic(&mut self) {
         self.finish_block(panic());
     }
 
-    /// Call a function that does not return.
+    // Call terminators
+    /// Call a function that does neither return nor unwind.
     pub fn call_noret(&mut self, ret: PlaceExpr, f: ValueExpr, args: &[ArgumentExpr]) {
         self.finish_block(Terminator::Call {
             callee: f,
             calling_convention: CallingConvention::C, // FIXME do not hard-code the C calling convention
             arguments: args.iter().copied().collect(),
             ret,
             next_block: None,
+            unwind_block: None,
         });
     }
 
-    // terminators with exactly 1 following block
-    pub fn call(&mut self, ret: PlaceExpr, f: ValueExpr, args: &[ArgumentExpr]) {
+    /// Call a function that does not unwind.
+    pub fn call_nounwind(&mut self, ret: PlaceExpr, f: ValueExpr, args: &[ArgumentExpr]) {
         let next_block = self.declare_block();
+        let next_block_kind = self.cur_block().kind;
         self.finish_block(Terminator::Call {
             callee: f,
             calling_convention: CallingConvention::C, // FIXME do not hard-code the C calling convention
             arguments: args.iter().copied().collect(),
             ret,
             next_block: Some(next_block),
+            unwind_block: None,
         });
-        self.set_cur_block(next_block)
+        self.set_cur_block(next_block, next_block_kind);
     }
 
-    /// Ignore unit type return value.
+    /// Call a function that does not unwind. Ignore unit type return value.
     pub fn call_ignoreret(&mut self, f: ValueExpr, args: &[ArgumentExpr]) {
         let next_block = self.declare_block();
+        let next_block_kind = self.cur_block().kind;
         self.finish_block(Terminator::Call {
             callee: f,
             calling_convention: CallingConvention::C, // FIXME do not hard-code the C calling convention
             arguments: args.iter().copied().collect(),
             ret: unit_place(),
             next_block: Some(next_block),
+            unwind_block: None,
+        });
+        self.set_cur_block(next_block, next_block_kind);
+    }
+
+    pub fn call(
+        &mut self,
+        ret: PlaceExpr,
+        f: ValueExpr,
+        args: &[ArgumentExpr],
+        unwind_block: BbName,
+    ) {
+        let next_block = self.declare_block();
+        let next_block_kind = self.cur_block().kind;
+        self.finish_block(Terminator::Call {
+            callee: f,
+            calling_convention: CallingConvention::C, // FIXME do not hard-code the C calling convention
+            arguments: args.iter().copied().collect(),
+            ret,
+            next_block: Some(next_block),
+            unwind_block: Some(unwind_block),
         });
-        self.set_cur_block(next_block);
+        self.set_cur_block(next_block, next_block_kind);
+    }
+
+    // terminator with 0 following blocks and 1 clean up block
```

Comment:
```suggestion
    /// terminator with 0 following blocks and 1 clean up block
```
Also please make this a proper sentence.

Reply:
In this file, before my changes, the Terminator was divided into three groups.
 

> // terminators with 0 following blocks

> // terminators with exactly 1 following block

> // terminators with 2 or more following blocks

I intended to create a new such group for `StartUnwind` with this comment. 
It might be better to put `StartUnwind` in the  `// terminators with exactly 1 following block`  group.

Therefore this comment was not meant to be a doc comment. 

So far only two terminators have a doc comment. Should I still add a doc comment to `StartUnwind` and `ResumeUnwind`?

Reply:
> Therefore this comment was not meant to be a doc comment.

Then there should be an empty line between the comment and the function. Otherwise it looks like it refers only to that function.

But yeah I also think this should be grouped with the "exactly 1 following block" terminators.